### PR TITLE
Stlx 9814 native token unit test

### PR DESCRIPTION
**Pull request by JoaquinTissera-Simtlix**
Add unit test for send token in Avaxc
Eth getToken util add filter for network since Avaxc has same address in testnet and mainnet. 

STLX-9814

### CODE REVIEW

File: modules/account-lib/test/unit/coin/avaxc/transferBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+import { TransferBuilder as AvaxCTransferBuilder } from '../../../../src/coin/avaxc';
+import { AvaxC, Eth } from '../../../../src';
+
+// import * as testData from '../../../resources/avaxc/avaxc';
+const amount = '20000';
+const toAddress = '0x7325A3F7d4f9E86AE62Cf742426078C3755730d5';
+const keyPair = new AvaxC.KeyPair();
+const key = keyPair.getKeys().prv as string;
+
+describe('ARC20 and AVAXERC20 Tokens', () => {
```
Comment (charly-simtlix) [inline modules/account-lib/test/unit/coin/avaxc/transferBuilder.ts:11]: ARC20?

File: modules/account-lib/test/unit/coin/avaxc/transferBuilder.ts [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+  describe('Maintest', () => {
+    it('PNG token transfer should succeed', async () => {
```
Comment (charly-simtlix) [inline modules/account-lib/test/unit/coin/avaxc/transferBuilder.ts:13]: it seems to be a lot of duplicated code. A plausible approach for this type of test is to use "parametric testing".
See https://medium.com/@nyablk97/parameterized-tests-with-jasmine-ecadb2856980 for a simple explanation

File: modules/account-lib/test/unit/coin/avaxc/transferBuilder.ts [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+      const result = builder.signAndBuild();
+      const decode = Eth.Utils.decodeTransferData(result);
+      should.equal(decode.amount, amount);
+      should.equal(decode.sequenceId, 2);
+      should.equal(decode.expireTime, 1590078260);
+    });
+  });
+
+  describe('Testnet', () => {
+    it('tPNG token transfer should succeed', async () => {
```
Comment (charly-simtlix) [inline modules/account-lib/test/unit/coin/avaxc/transferBuilder.ts:104]: Same as before:
it seems to be a lot of duplicated code. A plausible approach for this type of test is to use "parametric testing".
See https://medium.com/@nyablk97/parameterized-tests-with-jasmine-ecadb2856980 for a simple explanation

File: modules/account-lib/test/unit/coin/avaxc/transferBuilder.ts (hunk 2 above)
Comment (JoaquinTissera-Simtlix) [reply modules/account-lib/test/unit/coin/avaxc/transferBuilder.ts]: Done!

File: modules/account-lib/test/unit/coin/avaxc/transferBuilder.ts (hunk 1 above)
Comment (JoaquinTissera-Simtlix) [reply modules/account-lib/test/unit/coin/avaxc/transferBuilder.ts]: Remove

File: modules/account-lib/test/unit/coin/avaxc/transferBuilder.ts (hunk 3 above)
Comment (JoaquinTissera-Simtlix) [reply modules/account-lib/test/unit/coin/avaxc/transferBuilder.ts]: Done

**Review by charly-simtlix — APPROVED** [review]

File: modules/statics/src/account.ts [hunk 4]
```diff
@@ -74,6 +74,10 @@ export interface SolCoinConstructorOptions extends AccountConstructorOptions {
   tokenAddress: string;
 }
 
+export interface ARC20CoinContructorOptions extends Erc20ConstructorOptions {
+  assetIdXChain: string;
+}
+
```
Comment (sskim16) [inline modules/statics/src/account.ts:80]: ARC-20 no longer needed

File: modules/statics/src/account.ts [hunk 5]
```diff
@@ -254,6 +258,30 @@ export class SolCoin extends AccountCoinToken {
...
+ */
+export class ARC20Token extends AvaxERC20Token {
+  public assetIdXChain: string;
+  constructor(options: ARC20CoinContructorOptions) {
+    super(options);
+
+    this.assetIdXChain = options.assetIdXChain;
+  }
+}
+
```
Comment (sskim16) [inline modules/statics/src/account.ts:284]: ARC-20 no longer needed

File: modules/statics/src/account.ts [hunk 6]
```diff
@@ -888,3 +916,175 @@ export function tsolToken(
...
+      prefix,
+      suffix,
+      features,
+      decimalPlaces,
+      asset,
+      isToken: true,
+      primaryKeyCurve,
+    })
+  );
+}
```
Comment (sskim16) [inline modules/statics/src/account.ts:1047]: ARC-20 no longer needed

File: modules/statics/src/account.ts [hunk 7]
```diff
@@ -888,3 +916,175 @@ export function tsolToken(
...
+    contractAddress,
+    assetIdXChain,
+    asset,
+    features,
+    prefix,
+    suffix,
+    network,
+    primaryKeyCurve
+  );
+}
```
Comment (sskim16) [inline modules/statics/src/account.ts:1090]: ARC-20 no longer needed

File: modules/statics/src/coins.ts [hunk 8]
```diff
@@ -6,6 +6,10 @@ import {
   hederaCoin,
   stellarToken,
   talgoToken,
+  arc20Token,
+  tarc20Token,
```
Comment (sskim16) [inline modules/statics/src/coins.ts:10]: No more ARC-20 support

File: modules/statics/src/coins.ts [hunk 9]
```diff
@@ -1120,4 +1124,22 @@ export const coins = CoinMap.fromCoins([
...
+  avaxErc20('PNG', 'Pangolin', 18, '0x60781c2586d68229fde47564546784ab3faca982', UnderlyingAsset['avaxc:PNG']),
+  avaxErc20('USDT', 'Tether', 6, '0xc7198437980c041c805a1edcba50c1ce5db95118', UnderlyingAsset.USDT),
+  tavaxErc20('tQI', 'Test BenQi', 18, '0x8729438eb15e2c8b576fcc6aecda6a148776c0f5', UnderlyingAsset['avaxc:QI']),
+  tavaxErc20(
+    'tXAVA',
+    'Test Avalaunch',
+    18,
+    '0xd1c3f94de7e5b45fa4edbba472491a9f4b166fc4',
+    UnderlyingAsset['avaxc:XAVA']
+  ),
```
Comment (sskim16) [inline modules/statics/src/coins.ts:1140]: Nit: formatting 
tavaxErc20('tXAVA', 'Test Avalaunch', 18, '0xd1c3f94de7e5b45fa4edbba472491a9f4b166fc4', UnderlyingAsset['avaxc:XAVA'])

**Review by sskim16 — CHANGES_REQUESTED** [review]

File: modules/statics/src/coins.ts [hunk 10]
```diff
@@ -1120,4 +1124,22 @@ export const coins = CoinMap.fromCoins([
...
+  tavaxErc20('tKLO', 'Test Kalao', 18, '0xb27c8941a7df8958a1778c0259f76d1f8b711c35', UnderlyingAsset['avaxc:KLO']),
+  tavaxErc20('tJOE', 'Test Trader Joe', 18, '0x6e84a6216ea6dacc71ee8e6b0a5b7322eebc0fdd', UnderlyingAsset['avaxc:JOE']),
+  tavaxErc20('tPNG', 'Test Pangolin', 18, '0x60781c2586d68229fde47564546784ab3faca982', UnderlyingAsset['avaxc:PNG']),
+  tavaxErc20('tUSDT', 'Test Tether', 6, '0xc7198437980c041c805a1edcba50c1ce5db95118', UnderlyingAsset.USDT),
```
Comment (sskim16) [inline modules/statics/src/coins.ts:1144]: Nit: formatting on consistency
  tavaxErc20('tUSDT', 'Test Tether', 6, '0xc7198437980c041c805a1edcba50c1ce5db95118', UnderlyingAsset.USDT['avaxc:USDT']),

Comment (sskim16) [inline modules/statics/src/coins.ts:1144]: Nit: formatting on consistency
tavaxErc20('tUSDT', 'Test Tether', 6, '0xc7198437980c041c805a1edcba50c1ce5db95118', UnderlyingAsset.USDT['avaxc:USDT']),

**Review by sskim16 — DISMISSED** [review]

File: modules/account-lib/src/coin/eth/utils.ts [hunk 11]
```diff
@@ -479,10 +486,10 @@ export function getBufferedByteCode(methodId: string, rawData: string): Buffer {
  * @param tokenContractAddress The contract address to match against
  * @returns statics BaseCoin object for the matching token
  */
-export function getToken(tokenContractAddress: string): Readonly<BaseCoin> | undefined {
+export function getToken(tokenContractAddress: string, network : BaseNetwork): Readonly<BaseCoin> | undefined {
   const tokens = coins.filter((coin) => {
```
Comment (charly-simtlix) [inline modules/account-lib/src/coin/eth/utils.ts:490]: All expression can be simplified:

```
  const tokens = coins.filter((coin) => coin instanceof ContractAddressDefinedToken) &&
      coin.network === network && 
      coin.contractAddress.toLowerCase() === tokenContractAddress.toLowerCase())
```

**Review by charly-simtlix — APPROVED** [review]
There is one comment, but you can modify and go on

**Review by jrafael-simtlix — APPROVED** [review]

**Review by shiksb — APPROVED** [review]

File: modules/account-lib/src/coin/eth/utils.ts (hunk 11 above)
Comment (shiksb) [reply modules/account-lib/src/coin/eth/utils.ts:483]: thank you, I think it's ok to keep it for readability

**Pull request merged**